diff --git a/youtube-dl/youtube_dl/postprocessor/embedthumbnail.py b/youtube-dl/youtube_dl/postprocessor/embedthumbnail.py
index 3990908b6..30c313199 100644
--- a/youtube-dl/youtube_dl/postprocessor/embedthumbnail.py
+++ b/youtube-dl/youtube_dl/postprocessor/embedthumbnail.py
@@ -90,7 +90,7 @@ class EmbedThumbnailPP(FFmpegPostProcessor):

         elif info['ext'] in ['m4a', 'mp4']:
             atomicparsley = next((x
-                                  for x in ['AtomicParsley', 'atomicparsley']
+                                  for x in ['libatomicparsley.bin.so', 'AtomicParsley', 'atomicparsley']
                                   if check_executable(x, ['-v'])), None)

             if atomicparsley is None:
